src/components/About/WarningBanner.js

```diff
@@ -0,0 +1,77 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  Headline,
+  List,
+  MessageBanner,
+} from '@folio/stripes-components';
+
+import { isVersionCompatible } from '../../discoverServices';
+import css from './About.css';
+
+const WarningBanner = ({
+  interfaces,
+  modules,
+}) => {
+  const modulesArray = _.flatten(_.values(modules));
+  const allInterfaces = modulesArray.reduce((prev, curr) => Object.assign(prev, curr.okapiInterfaces), {});
+
+  const missingModules = Object.keys(allInterfaces).reduce((prev, curr) => {
+    if (!interfaces[curr]) {
+      return [...prev, `${curr} ${allInterfaces[curr]}`];
+    } else {
+      return prev;
+    }
+  }, []);
+
+  const incompatibleModule = Object.keys(allInterfaces).reduce((prev, curr) => {
+    if (interfaces[curr] && !isVersionCompatible(interfaces[curr], allInterfaces[curr])) {
+      return [...prev, `${curr} ${allInterfaces[curr]}`];
+    } else {
+      return prev;
+    }
+  }, []);
+
+  const missingModulesCount = _.compact(missingModules).length;
+  const incompatibleModulesCount = _.compact(incompatibleModule).length;
```

Comment:
@uladislausamets there is a lot going on here. This code will execute on every rerender. It would be nice to see if wrapping it in `useMemo` produces a better performance. I wonder if you tried to test it?